modules/extension/wmts/src/main/java/org/geotools/ows/wmts/response/WMTSGetCapabilitiesResponse.java

```diff
@@ -54,43 +50,38 @@ public WMTSGetCapabilitiesResponse(HTTPResponse response) throws ServiceExceptio
         this(response, null);
     }
 
+    /**
+     * Constructor that sets capabilities property. Input stream of response is closed immediately.
+     *
+     * @param response the httpResponse from the server
+     * @param hints not used
+     * @throws ServiceException thrown if server responds with ServiceException or a ill-formatted XML
+     * @throws IOException thrown if input stream is wrong
+     */
     public WMTSGetCapabilitiesResponse(HTTPResponse response, Map<String, Object> hints)
             throws ServiceException, IOException {
         super(response);
-
-        try {
-
-            Object object;
-            try (InputStream inputStream = response.getResponseStream()) {
+        Object object;
+        try (InputStream inputStream = response.getResponseStream()) {
```

Comment:
The two responses... don't they have a different content type, to start with?
You could use that to decide what to parse.

However, that assumes the server is setting the right content type, making the code less tolerant to non compliant servers... So not sure it's actually a good idea...

Reply:
The fix is for non compliant servers. That don't set content-type:application/vnd.ogc.se_xml.
If the content-type is set according to the spec, a superclass will throw the ServiceException.